pkg/resourcecollector/resourcecollector.go

```diff
@@ -394,19 +404,44 @@ func (r *ResourceCollector) prepareResourcesForCollection(
 		}
 
 		content := o.UnstructuredContent()
-		switch o.GetObjectKind().GroupVersionKind().Kind {
-		case "CouchbaseBackup", "CouchbaseBackupRestore":
-		default:
-			// Status shouldn't be retained when collecting resources
-			delete(content, "status")
+		crdList, err := storkops.Instance().ListApplicationRegistrations()
+		if err != nil {
+			return err
 		}
-		metadataMap := content["metadata"].(map[string]interface{})
-		// Remove all metadata except some well-known ones
-		for key := range metadataMap {
-			switch key {
-			case "name", "namespace", "labels", "annotations":
-			default:
-				delete(metadataMap, key)
+		resourceKind := o.GetObjectKind().GroupVersionKind().Kind
+		for _, crd := range crdList.Items {
+			for _, kind := range crd.Resources {
+				if kind.Kind == resourceKind {
+					// remove status from crd
+					if !kind.KeepStatus {
+						delete(content, "status")
+					}
+					// remove metadata annotations
+					metadataMap := content["metadata"].(map[string]interface{})
+					// Remove all metadata except some well-known ones
+					for key := range metadataMap {
+						switch key {
+						case "name", "namespace", "labels", "annotations":
+						default:
+							delete(metadataMap, key)
+						}
+					}
+					fields := strings.Split(kind.SuspendOptions.Path, ".")
+					if len(fields) > 1 {
```

Comment:
This isn't required here since it is used by ApplicationBackup and ApplicationClone too. Just doing it in the MigrationController is good